File: LightGlide/app/src/main/java/cy/ac/ucy/cs/epl341/team5/lightglide/db/model/Metrics.java
package cy.ac.ucy.cs.epl341.team5.lightglide.db.model;

import java.util.List;

public class Metrics {
    final int elevation;
    final int altitude;
    final List<Double> acceleration;
    final List<Double> gpsCoords;
    final double climb_descend_rate;

    Metrics(int elevation, int altitude, List<Double> acceleration, List<Double> gpsCoords, double climb_descend_rate) {
        this.elevation = elevation;
        this.altitude = altitude;
        this.acceleration = acceleration;
        this.gpsCoords = gpsCoords;
        this.climb_descend_rate = climb_descend_rate;
    }
}

File: LightGlide/app/src/main/java/cy/ac/ucy/cs/epl341/team5/lightglide/db/model/Metrics.kt
package cy.ac.ucy.cs.epl341.team5.lightglide.db.model

import android.os.Parcel
import android.os.Parcelable
import java.sql.Timestamp

data class Metrics constructor(
        val altitude: Double, val descendRate: Double,
        val timestamp: Timestamp, val gpsCoordinates: GPSCoordinates
) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readDouble(),
            parcel.readDouble(),
            Timestamp(parcel.readLong()),
            GPSCoordinates.createFromParcel(parcel)
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeDouble(altitude)
        parcel.writeDouble(descendRate)
        parcel.writeLong(timestamp.time)
        parcel.writeParcelable(gpsCoordinates, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Metrics> {
        override fun createFromParcel(parcel: Parcel): Metrics {
            return Metrics(parcel)
        }

        override fun newArray(size: Int): Array<Metrics?> {
            return arrayOfNulls(size)
        }
    }
}